### PR TITLE
Remove namespace.CacheEntry.GetConfig()

### PR DESCRIPTION
**Pull request by mmcshane**
**What changed?**
Remove namespace.CacheEntry.GetConfig()

**Why?**
Part of a larger refactoring to make CacheEntry immutable by prohibiting
access to its internal NamespaceInfo, NamespaceConfig, and
NamespaceReplicationConfig

**How did you test it?**
Existing unit tests

**Potential risks**

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/history/workflow/test_util.go [hunk 1]
```diff
@@ -37,11 +38,12 @@ import (
 func TestLocalMutableState(
 	shard shard.Context,
 	eventsCache events.Cache,
+	ns *namespace.CacheEntry,
 	logger log.Logger,
 	runID string,
 ) *MutableStateImpl {
 
-	msBuilder := NewMutableState(shard, eventsCache, logger, tests.LocalNamespaceEntry, time.Now().UTC())
+	msBuilder := NewMutableState(shard, eventsCache, logger, ns, time.Now().UTC())
```
Comment (yiminc) [inline service/history/workflow/test_util.go:46]: just curious why this change? is there case where we call this TestLocalMutableState() with cache entry other than tests.LocalNamespaceEntry?

File: service/history/workflow/util.go [hunk 2]
```diff
@@ -161,22 +164,21 @@ func TerminateWorkflow(
 // FindAutoResetPoint returns the auto reset point
 func FindAutoResetPoint(
 	timeSource clock.TimeSource,
-	badBinaries *namespacepb.BadBinaries,
+	ns VerifyBinaryChecksummer,
```
Comment (yiminc) [inline service/history/workflow/util.go:167]: It feel like taking a func here might be easier to read?
checksumVerifyFn func(string) error

**Review by yiminc — APPROVED** [review]

File: service/history/workflow/test_util.go (hunk 1 above)
Comment (mmcshane) [reply service/history/workflow/test_util.go:46]: Yes, that's exactly it.
